benchmark/indexing/benchmark_indexing.py

```diff
@@ -0,0 +1,173 @@
+"""
+A script to benchmark different methods for indexing BIDS datasets.
+"""
+
+import argparse
+import json
+import socket
+import subprocess
+import sys
+import tempfile
+import time
+from pathlib import Path
+
+try:
+    import bids
+
+    has_pybids = True
+except ModuleNotFoundError:
+    has_pybids = False
+
+try:
+    import ancpbids
+
+    has_ancpbids = True
+except ModuleNotFoundError:
+    has_ancpbids = False
+
+try:
+    import bids2table as b2t
+
+    has_bids2table = True
+except ModuleNotFoundError:
+    has_bids2table = False
+
+METHODS = ("pybids", "ancpbids", "bids2table")
+
+
+def du(path):
+    size_kb = subprocess.check_output(["du", "-sk", path])
+    size_kb = int(size_kb.split()[0].decode("utf-8"))
+    size_mb = int(size_kb) / 1000
+    return size_mb
+
+
+def benchmark_pybids(root: Path, workers: int):
+    assert has_pybids, "pybids not installed"
+    assert workers == 1, "pybids doesn't use multiple workers"
+    tic = time.monotonic()
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        indexer = bids.BIDSLayoutIndexer(
+            validate=False,
+            index_metadata=True,
```

Comment:
Out of curiosity can you try this w/ `index_metadata=False`? 

I suppose it would not be comparable since bids2table indexes metadata from JSON sidecars but I'd be curious how much this contributes to the slowness.